TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -79,341 +119,484 @@ namespace TEN::Entities::Creatures::TR3
+					else if (creature.Mood == MoodType::Bored)
+					{
+						if (Random::TestProbability(CIVVY_WAIT_CHANCE))
+							item.Animation.TargetState = CIVVY_STATE_IDLE;
+					}
+					else if ((creature.Enemy->HitPoints > 0 && creature.Enemy->Animation.Velocity.z < CIVVY_TARGET_ALERT_VELOCITY) &&
+							 (
+							  (creature.Enemy->IsLara() && AI.distance < CIVVY_ATTACK_CLOSE_PUNCH_RANGE ) ||
+							  (creature.Enemy->IsCreature() && AI.distance < SQUARE(BLOCK(0.5f)) )
+							 ))
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if ((creature.Enemy->IsLara() && AI.bite && AI.distance < CIVVY_ATTACK_WALKING_PUNCH_RANGE) ||
+							 (creature.Enemy->IsCreature() && creature.Enemy->HitPoints > 0 && AI.ahead && AI.distance < SQUARE(BLOCK(1))) )
+						item.Animation.TargetState = CIVVY_STATE_AIM_WALKING_PUNCH;
+					else if (AI.distance > CIVVY_WALK_RANGE)
+						item.Animation.TargetState = CIVVY_STATE_RUN;
+					else
+						item.Animation.TargetState = CIVVY_STATE_WALK;
 
-				break;
+					break;
 
-			case CIVVY_STATE_RUN_FORWARD:
-				creature->MaxTurn = CIVVY_RUN_TURN_RATE_MAX;
-				tilt = angle / 2;
+				case CIVVY_STATE_WAIT:
 
-				if (AI.ahead)
-					extraHeadRot.y = AI.angle;
+					creature.MaxTurn = 0;
+					
+					if (AI.ahead)
+						jointHeadRot.y = targetAngle;
+					else if (item.AIBits & GUARD)
+						jointHeadRot.y = AIGuard(&creature);
+
+					if (creature.Alerted || Random::TestProbability(CIVVY_WAIT_CHANCE))
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
 
-				if (item->AIBits & GUARD)
-					item->Animation.TargetState = CIVVY_WAIT;
-				else if (creature->Mood == MoodType::Escape)
-				{
-					if (Lara.TargetEntity != item && AI.ahead)
-						item->Animation.TargetState = CIVVY_STATE_IDLE;
 					break;
-				}
-				else if ((item->AIBits & FOLLOW) && (creature->ReachedGoal || laraAI.distance > SQUARE(SECTOR(2))))
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
-				else if (creature->Mood == MoodType::Bored)
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
-				else if (AI.ahead && AI.distance < CIVVY_WALK_RANGE)
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
 
-				break;
+				case CIVVY_STATE_RUN:
 
-			case CIVVY_AIM0:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
-				creature->Flags = 0;
+					creature.MaxTurn = CIVVY_RUN_TURN_RATE_MAX;
+					tilt = angle / 2;
 
-				if (AI.ahead)
-				{
-					extraTorsoRot.x = AI.xAngle;
-					extraTorsoRot.y = AI.angle;
-				}
+					if (AI.ahead)
+						jointHeadRot.y = AI.angle;
 
-				if (AI.bite && AI.distance < CIVVY_ATTACK0_RANGE)
-					item->Animation.TargetState = CIVVY_PUNCH0;
-				else
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
+					if (item.AIBits & GUARD)
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if (creature.Mood == MoodType::Escape)
+						item.Animation.TargetState = CIVVY_STATE_RUN;
+					//else if (creature.Mood == MoodType::Escape && Lara.TargetEntity != &item && AI.ahead)
```

Comment:
This is not commented in the official source, also you can use
`creature.Enemy != nullptr && creature.Enemy->IsLara() && GetLaraInfo(creature.Enemy)->TargetEntity != &item` to avoid nullptr

Reply:
Ah yes, this is one of my doubts.

This commented code is making that when prisoner is in escape mood, if Lara is ahead of him and is aiming at him, then he stops... Why is doing that?, What is the gameplay purpose? At first I though it got bugged. 

I couldn't check yet in TR3 because the part where there are prisoners, Lara hasn't got weapons. (And need weapons to do the weapons cheat).

I didn't want to fully remove the code because there may be a hidden purpose that I unknow, but meanwhile, to provide a working solution, I comented that feature, and let that in Escape mode he just run through the map (and in while running, he gets close to the target, then he stop escapping and pass to attack).

Reply:
Fine i will let the comment open, in case it would help later to avoid forgeting it.